### PR TITLE
File updates

### PR DESCRIPTION
**Pull request by JustCat80**
Recreated from #1273

Deprecates file for files

Also adds support for editing attachments within a message with editMessage

### CODE REVIEW

File: lib/Client.js [hunk 1]
```diff
@@ -609,7 +609,10 @@ class Client extends EventEmitter {
     * @arg {String} [content.messageReferenceID] [DEPRECATED] The ID of the message should be replied to. Use `messageReference` instead
     * @arg {Array<String>} [content.stickerIDs] An array of IDs corresponding to stickers to send
     * @arg {Boolean} [content.tts] Set the message TTS flag
-    * @arg {Object | Array<Object>} [file] A file object (or an Array of them)
+    * @arg {Array<Object>} [content.attachments] An array of attachment objects
+    * @arg {Buffer} content.attachments.file A buffer containing attachment data
+    * @arg {String} content.attachments.name What to name the attachment
+    * @arg {Object | Array<Object>} [file] [DEPRECATED] A file object (or an Array of them). Use `content.attachments` instead
```
Comment (bsian03) [inline lib/Client.js]: Alphabetical placement

File: lib/Client.js [hunk 2]
```diff
@@ -1248,6 +1257,9 @@ class Client extends EventEmitter {
     * @arg {Boolean} [content.allowedMentions.everyone] Whether or not to allow @everyone/@here.
     * @arg {Boolean | Array<String>} [content.allowedMentions.roles] Whether or not to allow all role mentions, or an array of specific role mentions to allow.
     * @arg {Boolean | Array<String>} [content.allowedMentions.users] Whether or not to allow all user mentions, or an array of specific user mentions to allow.
+    * @arg {Object | Array<Object>} [content.attachments] An array of attachment objects
```
Comment (bsian03) [inline lib/Client.js]: Object ain't an array

File: lib/Client.js [hunk 3]
```diff
@@ -1539,7 +1551,10 @@ class Client extends EventEmitter {
     * @arg {String} [options.components[].url] The URL that the component should open for users (type 2 style 5 only)
     * @arg {String} [options.content=""] A content string
     * @arg {Array<Object>} [options.embeds] An array of Discord embeds
-    * @arg {Object | Array<Object>} [options.file] A file object (or an Array of them)
+    * @arg {Object | Array<Object>} [options.attachments] An array of attachment objects
```
Comment (bsian03) [inline lib/Client.js]: Object ain't an array, also alphabetical ordering

File: lib/Client.js [hunk 4]
```diff
@@ -1539,7 +1551,10 @@ class Client extends EventEmitter {
...
+    * @arg {Buffer} options.attachments.file A buffer containing attachment data
+    * @arg {String} options.attachments.name What to name the attachment
+    * @arg {Object | Array<Object>} [options.file] [DEPRECATED] A file object (or an Array of them), use options.attachments instead
     * @arg {Buffer} options.file.file A buffer containing file data
     * @arg {String} options.file.name What to name the file
```
Comment (bsian03) [inline lib/Client.js]: Could probably just get rid of these 2

File: lib/Client.js [hunk 5]
```diff
@@ -1615,7 +1636,10 @@ class Client extends EventEmitter {
     * @arg {String} [options.components[].url] The URL that the component should open for users (type 2 style 5 only)
     * @arg {String} [options.content=""] A content string
     * @arg {Array<Object>} [options.embeds] An array of Discord embeds
-    * @arg {Object | Array<Object>} [options.file] A file object (or an Array of them)
+    * @arg {Object | Array<Object>} [options.attachments] An array of attachment objects
```
Comment (bsian03) [inline lib/Client.js]: Object ain't an array, also alphabetical ordering

File: lib/Client.js [hunk 6]
```diff
@@ -1615,7 +1636,10 @@ class Client extends EventEmitter {
...
+    * @arg {Buffer} options.attachments.file A buffer containing attachment data
+    * @arg {String} options.attachments.name What to name the attachment
+    * @arg {Object | Array<Object>} [options.file] [DEPRECATED] A file object (or an Array of them), use options.attachments instead
     * @arg {Buffer} options.file.file A buffer containing file data
     * @arg {String} options.file.name What to name the file
```
Comment (bsian03) [inline lib/Client.js]: Could just get rid of these 2

File: lib/Client.js [hunk 7]
```diff
@@ -1632,7 +1656,7 @@ class Client extends EventEmitter {
             tts: options.tts,
             allowed_mentions: this._formatAllowedMentions(options.allowedMentions),
             components: options.components
-        }, options.file).then((response) => options.wait ? new Message(response, this) : undefined);
+        }, options.attachments).then((response) => options.wait ? new Message(response, this) : undefined);
```
Comment (bsian03) [inline lib/Client.js]: Breaking deprecation compatibility

**Review by bsian03 — CHANGES_REQUESTED** [review]
Please make relevant changes on individual structure classes too

File: lib/Client.js [hunk 8]
```diff
@@ -1588,16 +1603,20 @@ class Client extends EventEmitter {
     * @arg {String} [options.components[].url] The URL that the component should open for users (type 2 style 5 only)
     * @arg {String} [options.content=""] A content string
     * @arg {Array<Object>} [options.embeds] An array of Discord embeds
-    * @arg {Object | Array<Object>} [options.file] A file object (or an Array of them)
-    * @arg {Buffer} options.file.file A buffer containing file data
-    * @arg {String} options.file.name What to name the file
+    * @arg {Object | Array<Object>} [options.file] [DEPRECATED] A file object (or an Array of them), use options.attachments instead
```
Comment (bsian03) [inline lib/Client.js]: Inconsistent deprecation message with createMessage (and other locations)

File: lib/Client.js [hunk 9]
```diff
@@ -1664,15 +1686,19 @@ class Client extends EventEmitter {
...
```
Comment (bsian03) [inline lib/Client.js]: Same here (ignore this if this gets included in the review, i tried to delete it and couldn't for some reason)

**Review by bsian03 — CHANGES_REQUESTED** [review]

File: lib/Client.js (hunk 8 above)
Comment (JustCat80) [reply lib/Client.js]: fixed

**Comment by JustCat80** [comment]
Content.attachments is actually used when editing a message, so either we should not use attachments, or we should make all of them a seperate argument, honestly I'm leaning towards the seperate argument

File: index.d.ts [hunk 10]
```diff
@@ -845,6 +847,7 @@ declare namespace Eris {
   }
   interface AdvancedMessageContent {
     allowedMentions?: AllowedMentions;
+    attachments?: {id: string, filename: string};
```
Comment (DonovanDMC) [inline index.d.ts]: this should be made into its own interface or separated out into different lines, it looks out of place as-is

that should also be a semicolon, not a comma
```suggestion
    attachments?: {
        id: string;
        filename: string
    };
```

Comment (JustCat80) [reply index.d.ts]: I didn't realize Attachment was already an interface lol

File: lib/structures/CommandInteraction.js [hunk 11]
```diff
@@ -357,10 +368,10 @@ class CommandInteraction extends Interaction {
     * @arg {Number} content.components[].type The type of component - If 1, it is a collection and a `components` array (nested) is required; if 2, it is a button; if 3, it is a select menu
     * @arg {String} [content.components[].url] The URL that the component should open for users (type 2 style 5 only)
     * @arg {String} [content.content=""] A content string
-    * @arg {Object} [content.embeds] An array of up to 10 embed objects. See [the official Discord API documentation entry](https://discord.com/developers/docs/resources/channel#embed-object) for object structure
```
Comment (bsian03) [inline lib/structures/CommandInteraction.js]: why was this removed

File: lib/structures/ComponentInteraction.js [hunk 12]
```diff
@@ -304,10 +315,10 @@ class ComponentInteraction extends Interaction {
     * @arg {Number} content.components[].type The type of component - If 1, it is a collection and a `components` array (nested) is required; if 2, it is a button; if 3, it is a select menu
     * @arg {String} [content.components[].url] The URL that the component should open for users (type 2 style 5 only)
     * @arg {String} [content.content=""] A content string
-    * @arg {Object} [content.embeds] An array of up to 10 embed objects. See [the official Discord API documentation entry](https://discord.com/developers/docs/resources/channel#embed-object) for object structure
```
Comment (bsian03) [inline lib/structures/ComponentInteraction.js]: same here

**Review by bsian03 — CHANGES_REQUESTED** [review]

File: lib/structures/CommandInteraction.js (hunk 11 above)
Comment (JustCat80) [reply lib/structures/CommandInteraction.js]: Honestly not sure, its back after pulling dev tho

**Comment by bsian03** [comment]
Due to the large amount of conflicts, will close and redo this PR at a future date. Note editMessage file change was merged already as part of #1285
